### DISCUSSION TITLE
Wsl from Microsoft Store not yet compatible with release of custom kernel versione 6.X (6.0,6.1,6.2-rc)

### DISCUSSION THREAD

**Discussion by WinSSitaly** [microsoft/WSL#9496] [General]
### Version

Microsoft Windows [Versione 10.0.19044.2364]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

6.0.19

### Distro Version

Ubuntu 22.04.1

### Other Software

my custom kernel for WSL 2 from my github
https://github.com/WinSSitaly/WSL-Linux-Kernel

### Repro Steps

after doing all update
sudo apt update && sudo apt upgrade && sudo apt autoremove && sudo apt dist-upgrade && sudo do-release-upgrade
Windows Subsystem for Linux Update - 5.10.102.2 (black penguin)
and then update wsl to version microsoft store
wsl -update
(blue penguin)
custom version of kernel 6.X cannot start more

### Expected Behavior

before update, using version CAB installer all was ok
after version from Microsoft Store error not start more

### Actual Behavior

after update to version from Microsoft Store error not start more

### Diagnostic Logs

not yet a way to log it

**Comment by WinSSitaly** [comment]
Error code: Wsl/Service/CreateInstance/0x80072746

**Comment by elsaco** [comment]
@WinSSitaly not using your kernel configs, `linux-6.1.6` boot okay on `WSL 1.0.3.0`. The config file from `WSL2-Linux-Kernel/Microsoft/config-wsl` was used and sanitized to build the new kernel. After `wsl --shutdown` the new kernel loads w/out errors:

```
toor@XPS:~$ uname -a
Linux XPS 6.1.6-microsoft-standard-WSL2 #1 SMP Mon Jan 16 09:20:37 PST 2023 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by WinSSitaly** [comment]
@elsaco what means "sanitized to build the new kernel"
i can restat from wslconfig 
https://github.com/microsoft/WSL2-Linux-Kernel/blob/linux-msft-wsl-5.15.y/arch/x86/configs/config-wsl
then have more than 20 new flag to set up
after that
make KCONFIG_CONFIG=config-wsl -j32
all was ok

i tink i set something Y connect to disable virtual disk so cannot boot my kernel config, now reset it all ok
